ios/storage.md

```diff
@@ -161,317 +161,620 @@ func downloadData() {
+
+Though it can be called anywhere in the app, we recommend that you instantiate the AWSS3TransferUtility in the ``appDidFinishLaunching`` lifecycle method. 
+
+### Manage a Transfer when a Suspended App Returns to the Foreground**
+
+When an app that has initiated a transfer becomes suspended and then returns to the foreground, the transfer may still be in progress or may have completed. In both cases, use the following code to re-establish the progress and completion handler blocks of the app.
+
+```swift
+let uploadTasks = transferUtility.getUploadTasks().result
+  for task in uploadTasks! {
+       task.setCompletionHandler(completionHandler!)
+       task.setProgressBlock(progressBlock!)
+}
+
+
+let downloadTasks = transferUtility.getDownloadTasks().result
+   for task in downloadTasks! {
+       task.setCompletionHandler(completionHandler!)
+       task.setProgressBlock(progressBlock!)
     }
 }
+
+
+let multiPartUploadTasks = transferUtility.getMultiPartUploadTasks().result
+    for task in multiPartUploadTasks! {
+        task.setCompletionHandler(completionHandler!)
+        task.setProgressBlock(progressBlock!)
+
+ }
+```
+
+### Transfer with Object Metadata
+
+The `AWSS3TransferUtilityUploadExpression` and `AWSS3TransferUtilityMultiPartUploadExpression` classes contain the method `setValue:forRequestHeader` where you can pass in metadata to Amazon S3. This example demonstrates passing in the Server-side Encryption Algorithm as a request header in uploading data to S3 using MultiPart.
+
+```swift
+
+let data: Data = Data() // The data to upload
+
+let uploadExpression = AWSS3TransferUtilityMultiPartUploadExpression()
+uploadExpression.setValue("AES256", forRequestHeader: "x-amz-server-side-encryption-customer-algorithm")
+uploadExpression.progressBlock = {(task, progress) in DispatchQueue.main.async(execute: {
+        // Do something e.g. Update a progress bar.
+    })
+}
+
+let transferUtility = AWSS3TransferUtility.default()
+
+transferUtility.uploadUsingMultiPart(data:data,
+            bucket: "S3BucketName",
+            key: "S3UploadKeyName",
+            contentType: "text/plain",
+            expression: uploadExpression,
+            completionHandler: nil).continueWith { (task) -> AnyObject! in
+                if let error = task.error {
+                    print("Error: \(error.localizedDescription)")
+                }
+
+                return nil;
+            }
 ```
 
-**Response Mapping Template**
+### Transfer with Access Control List
+
+To upload a file and specify permissions for it, you can use predefined grants, also known as canned ACLs. The following code shows you how to setup a file with publicRead access using the AWSS3 client.
```

Comment:
This is a pretty complex feature and arguably I'm not sure how good this is in a Mobile app as our "front page" in documentation, but let's discuss. If we do keep this we should link to the S3 docs: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html

Reply:
Link added. Lets chat on whether to keep or remove.

Reply:
Removed as discussed.